### PR TITLE
feat(legacy): `InputTime` & `InputDateTime` support `AM` / `PM` formats

### LINKED ISSUE #1229: 🚀 - Support 12h am/pm input in InputTime
### Which @taiga-ui/* package(s) are relevant/releated to the feature request?

kit

### Description

Support 12h am/pm input in InputTime.

Some ideas: 
* minimal solution is probably to allow setting max to 12:00, then one can manually add an am/pm toggle or radio input
* nicer solution would be to allow setting the mode to 12h, and then am/pm toggle is shown and handled under the hood
  * instead of a toggle or radio input for am/pm, it could also be part of the masked input, so one would type e.g. `09:00 am`

Related to https://github.com/Tinkoff/taiga-ui/issues/70, https://github.com/Tinkoff/taiga-ui/issues/270.

### PR DESCRIPTION
**Pull request by nsbarsukov**
Fixes #1229

### CODE REVIEW

File: projects/cdk/date-time/test/time.spec.ts [hunk 1]
```diff
@@ -127,6 +127,36 @@ describe('TuiTime', () => {
...
                 expect(time.ms).toBe(888);
             });
+
+            describe('mode with AM / PM', () => {
+                (
+                    [
+                        ['12:00 AM', {hours: 0, minutes: 0}],
+                        ['12:34 AM', {hours: 0, minutes: 34}],
+                        ['12:59 AM', {hours: 0, minutes: 59}],
+                        ['12:59 AM', {hours: 0, minutes: 59}],
```
Comment (mdlufy) [inline projects/cdk/date-time/test/time.spec.ts:137]: Duplication?

File: projects/cdk/date-time/time.ts [hunk 2]
```diff
@@ -134,6 +135,22 @@ export class TuiTime implements TuiTimeLike {
         );
     }
 
+    private static parseHours(time: string): number {
+        const hours = Number(time.slice(0, 2));
+        const [meridiem = ''] =
+            /[ap]m/.exec(time.toLowerCase().replaceAll(/\W/g, '')) || [];
```
Comment (mdlufy) [inline projects/cdk/date-time/time.ts:141]: Maybe take it out to variable with "talking name" such complicated regexp? What do you think?

File: projects/cdk/date-time/test/time.spec.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/cdk/date-time/test/time.spec.ts]: ✅

File: projects/cdk/date-time/time.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/cdk/date-time/time.ts]: I've created new static method `parseMeridiemPeriod`

Comment (nsbarsukov) [reply projects/cdk/date-time/time.ts]: ✅

**Review by mdlufy — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**